jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java

```diff
@@ -704,7 +724,193 @@ protected void deployScriptsToUserJwalaScriptsDir(Jvm jvm, User user) throws Com
         }
 
     }
+    
+    public Jvm upgradeJDKAndDeployJvm(String jvmName, User user) {
+        boolean didSucceed = false;
+        Jvm jvm = getJvm(jvmName);
+        LOGGER.debug("Start upgradeJDKAndDeployJvm for {} by user {}", jvmName, user.getId());
+
+        historyFacadeService.write(jvm.getHostName(), jvm.getGroups(), "Starting to upgrade JDK :: JVM " +
+                jvm.getJvmName() + " on host " + jvm.getHostName(), EventType.USER_ACTION_INFO, user.getId());
+
+        //add write lock for multiple write
+        binaryDistributionLockManager.writeLock(jvmName + "-" + jvm.getId().toString());
+
+        try {
+            if (jvm.getState().isStartedState()) {
+                final String errorMessage = "The remote JVM " + jvm.getJvmName() + " must be stopped before attempting to generate the JVM";
+                LOGGER.info(errorMessage);
+                throw new InternalErrorException(FaultType.REMOTE_COMMAND_FAILURE, errorMessage);
+            }
+            //Does this required??
+            validateJvmAndAppResources(jvm);
+
+            checkForJvmBinaries(jvm);
+
+            distributeBinaries(jvm);
+            
+            //This will update the Setenv file with upgraded JDK
+            deployJvmResourceFilesForJDKUpgrade(jvm, user);

```

Comment:
This deploys all of the resources for the JVM. You only want to deploy the setenv.bat or setenv.sh that defines the JAVA_HOME.

Also, when does that file get updated? I would expect it to be updated as part of this method before it is deployed.

Reply:
deployJvmResourceFilesForJDKUpgrade() is a newly created method. The setenv file update is handled in this method